src/test/scala/com/fulcrumgenomics/umi/DuplexConsensusCallerTest.scala

```diff
@@ -0,0 +1,265 @@
+    //    First pair is good
+    //    Second pair is good
+    builder.addPair(name="ba1", start1=200, start2=100, strand1=Minus, strand2=Plus, attrs=Map(MI -> "foo/B")).foreach { r =>
+      r.setReadString(if (r.getFirstOfPairFlag) "CCCCCCCCCC" else "AAAAAAAAAA")
+    }
+    builder.addPair(name="ba2", start1=200, start2=100, strand1=Minus, strand2=Plus, attrs=Map(MI -> "foo/B")).foreach { r =>
+      r.setReadString(if (r.getFirstOfPairFlag) "CCCCCCCCCC" else "AAAAAAAAAA")
+    }
+
+    val recs = caller(q=20).consensusReadsFromSamRecords(builder.toSeq)
+    recs should have size 2
+    val r1 = recs.find(_.getFirstOfPairFlag).getOrElse(fail("R1 missing."))
+    val r2 = recs.find(_.getSecondOfPairFlag).getOrElse(fail("R2 missing."))
+
+    { // Check the per-read tags
+      import ConsensusTags.PerRead._
+      r1.getIntegerAttribute(RawReadCount)      shouldBe 5
+      r1.getIntegerAttribute(AbRawReadCount)    shouldBe 3
+      r1.getIntegerAttribute(BaRawReadCount)    shouldBe 2
+      r1.getIntegerAttribute(MinRawReadCount)   shouldBe 4 // where the no-call or low-qual base are
+      r1.getIntegerAttribute(AbMinRawReadCount) shouldBe 2 // where the no-call or low-qual base are
+      r1.getIntegerAttribute(BaMinRawReadCount) shouldBe 2
+      r1.getFloatAttribute(RawReadErrorRate)    shouldBe (1 / 48f) // 50 bases, minus 2 no/low-qual bases = 48
+      r1.getFloatAttribute(AbRawReadErrorRate)  shouldBe (1 / 28f)
+      r1.getFloatAttribute(BaRawReadErrorRate)  shouldBe 0f
+
+      r2.getIntegerAttribute(RawReadCount)      shouldBe 5
+      r2.getIntegerAttribute(AbRawReadCount)    shouldBe 3
+      r2.getIntegerAttribute(BaRawReadCount)    shouldBe 2
+      r2.getIntegerAttribute(MinRawReadCount)   shouldBe 5
+      r2.getIntegerAttribute(AbMinRawReadCount) shouldBe 3
+      r2.getIntegerAttribute(BaMinRawReadCount) shouldBe 2
+      r2.getFloatAttribute(RawReadErrorRate)    shouldBe (1 / 50f)
+      r2.getFloatAttribute(AbRawReadErrorRate)  shouldBe (1 / 30f)
+      r2.getFloatAttribute(BaRawReadErrorRate)  shouldBe 0f
+    }
+
+    { // Check the per-base tags
+      import ConsensusTags.PerBase._
+      r1.getSignedShortArrayAttribute(AbRawReadCount)  shouldBe Array[Byte](3,3,2,2,3,3,3,3,3,3)
+      r1.getSignedShortArrayAttribute(AbRawReadErrors) shouldBe Array[Byte](0,0,0,0,1,0,0,0,0,0)
+      r1.getSignedShortArrayAttribute(BaRawReadCount)  shouldBe Array[Byte](2,2,2,2,2,2,2,2,2,2)
+      r1.getSignedShortArrayAttribute(BaRawReadErrors) shouldBe Array[Byte](0,0,0,0,0,0,0,0,0,0)
+      r2.getSignedShortArrayAttribute(AbRawReadCount)  shouldBe Array[Byte](3,3,3,3,3,3,3,3,3,3)
+      r2.getSignedShortArrayAttribute(AbRawReadErrors) shouldBe Array[Byte](0,0,0,0,1,0,0,0,0,0)
+      r2.getSignedShortArrayAttribute(BaRawReadCount)  shouldBe Array[Byte](2,2,2,2,2,2,2,2,2,2)
+      r2.getSignedShortArrayAttribute(BaRawReadErrors) shouldBe Array[Byte](0,0,0,0,0,0,0,0,0,0)
+    }
```

Comment:
Yup.  The fact that the R strand stuff gets reversed/RC's is a bit PITA for test writing.